bugsnag/utils.py

```diff
@@ -316,3 +317,48 @@ def get(self):
 
     def set(self, new_value):
         setattr(ThreadContextVar.local_context(), self.name, new_value)
+
+
+# to_rfc3339: format a datetime instance to match to_rfc3339/iso8601 with
+# milliseconds precision
+# Python can do this natively from version 3.6, but we need to include a
+# fallback implementation for Python 3.5
+try:
+    # this will raise if 'timespec' isn't supported
+    datetime.utcnow().isoformat(timespec='milliseconds')  # type: ignore
+
+    def to_rfc3339(dt: datetime) -> str:
+        return dt.isoformat(timespec='milliseconds')  # type: ignore
+
+except Exception:
+    def _get_timezone_offset(dt: datetime) -> str:
+        if dt.tzinfo is None:
+            return ''
+
+        utc_offset = dt.tzinfo.utcoffset(dt)
+
+        if utc_offset is None:
+            return ''
+
+        sign = '+'
+
+        if utc_offset.days < 0:
+            sign = '-'
+            utc_offset = -utc_offset
+
+        hours_offset, minutes = divmod(utc_offset, timedelta(hours=1))
+        minutes_offset, seconds = divmod(minutes, timedelta(minutes=1))
+
+        return '{}{:02}:{:02}'.format(sign, hours_offset, minutes_offset)
+
+    def to_rfc3339(dt: datetime) -> str:
+        return '{:04}-{:02}-{:02}T{:02}:{:02}:{:02}.{:03}{}'.format(
```

Comment:
The stdlib implementation casts each component to int (`{:02d}`, etc), is that necessary here?

Reply:
`d` is the default for integers, but I guess it's possibly a little safer to specify it explicitly